src/js/components/Header.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Link, HashRouter } from 'react-router-dom';
+
+class Header extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      authenticated: false
+    };
+    this.logout = this.logout.bind(this);
+  }
+  /**
+   * @memberof Header logs out the user when invoked
+   */
+  logout() {
```

Comment:
Define class methods as arrow functions to avoid using `bind(this)` in the constructor. Simply do: 
```
logout = () => {
   // define your logout function here
}
```